raco/backends/myria/myria.py

```diff
@@ -1268,7 +1268,16 @@ def fire(self, exp):
             return MyriaLimit(exp.count,
                               algebra.Collect(
                                   MyriaLimit(exp.count, exp.input)))
+        return exp
+
+
+class GlobalOrderBy(rules.Rule):
 
+    def fire(self, exp):
+        if exp.__class__ == algebra.OrderBy:
+            return MyriaInMemoryOrderBy(
```

Comment:
The order in which the `Collect` operator receives tuples is immaterial. The local `OrderBy` is just there in order to apply `Limit`. The only user-visible ordering takes place in the final, global `OrderBy` on the coordinator.

Actually, I'm not sure now that either the existing `CollectBeforeLimit` or the new `GlobalOrderBy` rules will properly transform a `Limit(OrderBy)` logical expression. The transformation should be `Limit(OrderBy) ->  Limit(OrderBy(Collect(Limit(OrderBy))))`, but `CollectBeforeLimit` would transform it into `Limit(Collect(Limit(OrderBy))` (which is clearly incorrect), and your version of `GlobalOrderBy` (along with `CollectBeforeLimit`, which is evaluated first) would transform it like `Limit(OrderBy) -> Limit(Collect(Limit(OrderBy)) -> Limit(Collect(Limit(OrderBy(Collect(input)))))`, which would presumably be collapsed into `Limit(OrderBy(Collect(input)))`, since consecutive `Limit`s and `Collect`s are no-ops. This final transformation is actually correct, but highly suboptimal, since it omits the local `Limit` and `OrderBy`, forcing the entire original input to be streamed to the coordinator, which defeats the purpose of these local/global decomposition rules.